src/execution_plan/optimizations/traverse_order.c

```diff
@@ -107,23 +107,23 @@ static bool _valid_arrangement(const Arrangement arrangement, uint exps_count) {
 	 * exp2: [L1]
 	 * Isn't valid, as currently the first expression is converted
 	 * into a scan operation. */
-	if((exp->src_node->label || exp->dest_node->label) &&
+	QGNode *src = QueryGraph_GetNodeByAlias(qg, exp->src); // TODO unwisely expensive
+	QGNode *dest = QueryGraph_GetNodeByAlias(qg, exp->dest); // TODO unwisely expensive
+	if((src->label || dest->label) &&
 	   exp->edge &&
 	   exp->operand_count == 1) return false;
 
 	for(int i = 1; i < exps_count; i++) {
 		exp = arrangement[i];
-		QGNode *src = exp->src_node;
-		QGNode *dest = exp->dest_node;
 		int j = i - 1;
 
 		// Scan previous expressions.
 		for(; j >= 0; j--) {
 			AlgebraicExpression *prev_exp = arrangement[j];
-			if(prev_exp->src_node == src ||
-			   prev_exp->dest_node == src ||
-			   prev_exp->src_node == dest ||
-			   prev_exp->dest_node == dest) break;
+			if(prev_exp->src == exp->src     ||
+			   prev_exp->dest == exp->src    ||
+			   prev_exp->src == exp->dest    ||
+			   prev_exp->dest == exp->dest) break;
```

Comment:
Let's use our own version of `strcmp` which perform pointer comparison first
this is a general comment, for any case where we perform `==` between two string pointers instead of string comparison